### PR TITLE
fix: added github and gitlab response mock and replaced  external calls

### LINKED ISSUE #9287: Use mocks for ApplicationSet Gitea, GitHub, and GitLab unit tests
# Summary

The Gitea unit tests for the PR and SCM generators currently make external API calls. This makes the tests flaky. We should instead use a mocked service.

https://github.com/argoproj/argo-cd/blob/master/applicationset/services/scm_provider/gitea_test.go
https://github.com/argoproj/argo-cd/blob/master/applicationset/services/pull_request/gitea_test.go

# Motivation

Last week integration tests were broken for a couple days because of a change in the tested Gitea org.

# Proposal

We could follow the model of the Bitbucket tests and use a mocked service.

https://github.com/argoproj/argo-cd/blob/1d765a072cfdf3921cf7bf43c353bda61d57347c/applicationset/services/pull_request/bitbucket_server_test.go#L14

### PR DESCRIPTION
**Pull request by rishabh625**
Fixes  #9287 part of

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by rishabh625** [comment]
@leoluz @jannfis : Can you please review this??

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Looks great!

Can you look at adding coverage back for a 404 in the github SCM provider?

image

File: test/container/Procfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-notification: sh -c "FORCE_LOG_COLORS=4 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_TLS_DATA_PATH=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} ARGOCD_BINARY_NAME=argocd-notifications go run ./cmd/main.go --loglevel debug"
\ No newline at end of file
+applicationset-controller: [ "$BIN_MODE" == 'true' ] && COMMAND=./dist/argocd || COMMAND='go run ./cmd/main.go' && sh -c "FORCE_LOG_COLORS=4 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_TLS_DATA_PATH=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} ARGOCD_ASK_PASS_SOCK=/tmp/applicationset-ask-pass.sock ARGOCD_SSH_DATA_PATH=${ARGOCD_SSH_DATA_PATH:-/tmp/argocd-local/ssh} ARGOCD_BINARY_NAME=argocd-applicationset-controller $COMMAND --loglevel debug --metrics-addr localhost:12345 --probe-addr localhost:12346 --argocd-repo-server localhost:${ARGOCD_E2E_REPOSERVER_PORT:-8081}"
+notification: sh -c "FORCE_LOG_COLORS=4 ARGOCD_FAKE_IN_CLUSTER=true ARGOCD_TLS_DATA_PATH=${ARGOCD_TLS_DATA_PATH:-/tmp/argocd-local/tls} ARGOCD_BINARY_NAME=argocd-notifications go run ./cmd/main.go --loglevel debug"
```
Comment (leoluz) [inline test/container/Procfile]: I wonder if we can get rid of this `test/container/Procfile` and just use the main one? 
@alexmt thoughts?

**Comment by rishabh625** [comment]
> Looks great!
> 
> Can you look at adding coverage back for a 404 in the github SCM provider?
> 
> image

Yeah I should that takes the coverage down

File: test/container/Procfile (hunk 1 above)
Comment (rishabh625) [reply test/container/Procfile]: @alexmt : please comment

**Review by crenshaw-dev — APPROVED** [review]
lgtm besides Leo's question. Thanks again, awesome work!

File: test/container/Procfile (hunk 1 above)
Comment (rishabh625) [reply test/container/Procfile]: @leoluz : I think lets keep this file as it is, may be we might have case that a component is not required in e2e but have to run when we do make start

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.4 for 2.4.12.

**Pull request merged**